### ISSUE TITLE
Nomad exposing address field when running TLS for RPC. Similar also showing for consul checks

### ISSUE THREAD

**Issue by Cryson** [jippi/hashi-ui#231]
I just enabled TLS in nomad and I am getting a ton of TLS errors now in hashi-ui for RPC. What is happening is hashi-ui is trying to validate the nomad TLS certificate against the IP address being exposed by nomad which is invalid, see errors below:

`hashi-ui-linux-amd64: ERROR #033[0m [ClusterStatistics] Unable to fetch node stats for blah (Get http://127.0.0.1:4646/v1/node/02d76028-228e-02bb-4083-3d0385944f4d?region=cloudbolt&wait=60000ms: malformed HTTP response "\x15\x03\x01\x00\x02\x02")`

Is there any upcoming release to fix this? I've seen the skip_verify in dev, and if that's the solution I agree with that since we're tied to what nomad is exposing at the moment. If we agree, any idea on when a new release with skip verify will be released?

**Comment by multipathmaster** [comment]
I would like to see this figured out as well.

**Comment by jippi** [comment]
maybe #194 is relevant?

**Comment by Cryson** [comment]
Agreed, I found the same the further I looked into it.

I have opened a similar issue with nomad that stems from the same IP exposure issue  [nomad #2568](https://github.com/hashicorp/nomad/issues/2568) which might prompt the need for nomad to expose dns fqdn names or node names rather than IP's, or give us the option in nomad.conf to do so.

Closing this issue as it correlates with a nomad issue and not hashi-ui.
